### DISCUSSION TITLE
Option to check if the item can be delivered based on location of the customer

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [vendurehq/vendure#1843] [Q&A] [answered]
Looking for a hook to listen to AddItemToOrder and run a check and fail the item to be added

**Comment by michaelbromley** [comment, marked as answer]
I suggest you override the `addItemToOrder` mutation to implement this - there's no other way to do it in such a way that you can cause the item not to be added if it fails some custom validation check.

See https://www.vendure.io/docs/plugins/plugin-examples/extending-graphql-api/#override-built-in-resolvers
